src/hid/audio.cpp

```diff
@@ -387,7 +387,7 @@ void AudioHandle::Impl::InternalCallback(int32_t* in, int32_t* out, size_t size)
                 break;
             default: break;
         }
-        cb((const float**)fin, (float**)fout, size / 2);
+        cb((InputBuffer)fin, (OutputBuffer)fout, size / 2);
```

Comment:
Is this cast really necessary?

Reply:
So the way we have it now fin is declared as `fin float[size]`, which if I'm not mistaken allocates the space at compile time (maybe not since it depends on size to know how much space...). I'm not sure of a way to do this with our typedef without doing something like `InputBuffer fin = malloc(size)`

Reply:
```cpp
// The type of fin and fout is essentially float**, except that they're 
// both stack-allocated and thus local to the function.
float* fin[chns];
float* fout[chns];
// This should compile without problems
cb(fin, fout, size / 2);
```

Reply:
I agree, I think the cast is unnecessary here. (I had a redundant cast for `float**` in the previous code. So that's probably why it's here).